contrib/kubespray/script/openpai-generator.py

```diff
@@ -153,13 +154,51 @@ def get_kubernetes_node_info_from_API():
     return ret
 
 
+def get_pai_daemon_resource_request(cfg):
+    ret = {
+        "cpu-resource": 0,
+        "mem-resource": 0
+    }
+    if "qos-switch" not in cfg or cfg["qos-switch"] == "false":
+        logger.info("Ignore calculate pai daemon resource usage since qos-switch set to false")
+        return ret
+
+    pai_daemon_services = ["node-exporter", "job-exporter", "log-manager"]
```

Comment:
This should include all pai service

Reply:
That's all PAI services which will deployed in worker node. Others are 3rd-party services and will calculated by:
https://github.com/microsoft/pai/blob/364aee769a421771ab76284d55fd7039767d3f20/contrib/kubespray/script/openpai-generator.py#L145-L148

`dshuttle` is not released, we don't include it

Reply:
How we mantain this list? How new service owner knows he should add this service here?
Why not just check all services on worker, for both PAI and 3rd party? In case they specified resource

Reply:
The reason is:  At this point, the service not deployed in worker. So we need to parse source code to get the resource.

Reply:
Deploy storage manager only on master instead of worker on int bed.

Reply:
Will do it later